File: nestjs_drizzle/src/security/jwt-verification.exception.ts
/**
 * JWT 검증 실패 예외 (SR §4). 실패 원인별 resultCode 를 담는다.
 * 인증 미들웨어가 이를 catch 하여 HTTP 200 표준 응답으로 변환한다.
 */
export class JwtVerificationException extends Error {
  constructor(readonly resultCode: number) {
    super(`JwtVerificationException(resultCode=${resultCode})`);
    this.name = 'JwtVerificationException';
  }
}

File: spring_mvc_jpa_parallel_virtual_thread/src/main/java/me/yeonhyuk/heavy_traffic/spring_mvc_jpa_parallel_virtual_thread/security/JwtVerificationException.java
package me.yeonhyuk.heavy_traffic.spring_mvc_jpa_parallel_virtual_thread.security;

import lombok.Getter;
import me.yeonhyuk.heavy_traffic.spring_mvc_jpa_parallel_virtual_thread.common.ResultCodeEnum;

/**
 * JWT 검증 실패 시 발생. 어떤 실패인지에 해당하는 resultCode 를 담는다 (SR §4).
 */
@Getter
public class JwtVerificationException extends RuntimeException {

    private final ResultCodeEnum resultCode;

    public JwtVerificationException(ResultCodeEnum resultCode) {
        super(resultCode.name());
        this.resultCode = resultCode;
    }
}
